src/Database/Esqueleto/Experimental.hs

```diff
@@ -379,7 +380,8 @@ data SqlSetOperation a =
   | UnionAll (SqlSetOperation a) (SqlSetOperation a)
   | Except (SqlSetOperation a) (SqlSetOperation a)
   | Intersect (SqlSetOperation a) (SqlSetOperation a)
-  | SelectQuery (SqlQuery a)
+  | SelectQuery  (SqlQuery a)
+  | SelectQueryP (SqlQuery a)
```

Comment:
Does it make sense to use something like `SelectQuery NeedsParens (SqlQuery a)` instead of this?

Reply:
i think it would require breaking the public API since SelectQuery is used directly. Maybe PatternSynonyms could fix that?

Reply:
Oh shoot! That's nasty. 

Actually, since this is adding a constructor to a public type, that's a breaking change too... If we'd only exposed named functions then we could get away with adding constructors for internal use.

Maybe let's add a comment that users should only be *constructing* with these, and not pattern matching? And additional constructors may be added without a major version bump?

It's an `Experimental` module so I'm not too particular about breaking changes.

Reply:
If we're gonna break the API then I'd rather break it properly and hide the constructors completely and replace them with smart constructors.

Reply:
Same. 

Hm. Let's put deprecation notices on the constructors, so people get warns with "how to fix" and not breaks. Then we can hide them in another release and remove the deprecation notices.

Alternatively, we move the constructors to an `Internal` module, make pattern synonyms, warn on the pattern synonym, and point to the named functions. Then a future release can delete the pattern synonyms.

Reply:
If we go the pattern synonyms route there isn't really a reason to deprecate the API. Not like there are going to be new set operations. It's just gonna be implementation details that potentially change.

Reply:
Good point